### PR TITLE
LPS-195690 Customer - License Selection for DXP Apps - Trial

### PR DESCRIPTION
**Pull request by ocesar9**
https://liferay.atlassian.net/browse/LPS-195690

### CODE REVIEW

**Comment by juliocezar96** [comment]
Just starting reviewing :)

**Comment by juliocezar96** [comment]
Just starting reviewing :)

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialMethod.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import timeline from '../../../../assets/images/timeline.png';
+
+import './TrialMethod.scss';
+
+interface TrialTimelineProps {
+  sku?: SKU[];
+}
+
+export function TrialMethod({sku}: TrialTimelineProps) {
+	console.log(sku);
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialMethod.tsx:19]: Please remove it

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialMethod.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+		'April',
+		'May',
+		'June',
+		'July',
+		'August',
+		'September',
+		'October',
+		'November',
+		'December',
+	];
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialMethod.tsx:33]: Shouldn't we move it to a generic components folder?

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialMethod.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+	const date = new Date();
+
+	date.setTime(date.getTime() + 30 * 24 * 60 * 60 * 1000);
+
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialMethod.tsx:38]: You may want to create a function to describe this calculation

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialTimeline.scss [hunk 4]
```diff
@@ -1,53 +1,32 @@
...
-			height: 8px;
-			margin-left: 3px;
-			position: relative;
-			top: -4px;
-			width: 8px;
-		}
-	}
+.trial-timeline {
+  display: flex;
+  flex-direction: column;
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialTimeline.scss:3]: Please use 'd-flex flex-column' Bootstrap here. Same for 'w-100' and 'border-radius'.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialMethod.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+/* eslint-disable react/no-unescaped-entities */
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialMethod.tsx:6]: Is it really necessary?

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialTimeline.tsx [hunk 6]
```diff
@@ -27,57 +27,29 @@ export function TrialTimeline() {
...
+        }
+        iconLeft={false}
+        onClick={() => {
+          setHandleSelected(true);
+          onSelectLicense(true);
+        }}
+        selected={handleSelected ? true : false}
+        title="Trial License"
+      />
+    </div>
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/GetAppModal/TrialTimeline.tsx:53]: We already have the <ClayCard/> component. Please refactor this using ClayCard. You can use the 'active' _className_ plus the onClick().

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/licenseCard/index.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+import ClayIcon from "@clayui/icon";
+
+import "./LicenseCard.scss";
+
+import ClayButton from "@clayui/button";
+
+import infoCircleFullIcon from "../../../../assets/icons/icon_info_circle_full.svg";
+
+const LicenseSectorCard: React.FC<any> = ({
+	icon = <img alt="Info" src={infoCircleFullIcon} />,
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/licenseCard/index.tsx:15]:
```suggestion
import infoCircleFullIcon from "../../../../assets/icons/icon_info_circle_full.svg";

const pageRedirectLink = window.location.href = origin;

const LicenseSectorCard: React.FC<any> = ({
	icon = <img alt="Info" src={infoCircleFullIcon} />,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/licenseCard/index.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+				</span>
+				<div className="align-items-center d-flex license__card__buttons__container p-1">
+					<span>
+						<ClayButton
+							aria-label=""
+							className="align-items-center d-flex justify-content-center license__card__buttons p-2"
+							displayType="primary"
+							onClick={() => {
+								window.location.href = origin;
+							}}
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/licenseCard/index.tsx:42]:
```suggestion
							onClick={() => {
								pageRedirectLink;
							}}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/licenseCard/index.tsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+				</div>
+			</div>
+
+			<div className="d-flex flex-column license__card__tier mt-4 p-4">
+				<div className="font-weight-bold license__card__tier__title mb-1">
+					License Prices
+				</div>
+
+				{licensetiers.map((license: any, index: any) => (
+					<span className="license__card__tier__price__text" key={index}>
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/licenseCard/index.tsx:79]: Please type it correctly.

**Pull request closed without merging**